pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -240,9 +240,9 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 		externalNameSvcInstanceMap: make(map[host.Name][]*model.ServiceInstance),
 		workloadInstancesIndex:     workloadinstances.NewIndex(),
 		initialSyncTimedout:        atomic.NewBool(false),
-		networkManager:             initNetworkManager(options),
 		configCluster:              options.ConfigCluster,
 	}
+	c.networkManager = initNetworkManager(c, options)
```

Comment:
I might be missing context here. Do we no longer need to start gatewayResourceClient?

Reply:
The delayedinformerclient already starts it. Forgot to mention that

Reply:
Thanks